model/client.go

```diff
@@ -1457,6 +1457,16 @@ func (c *Client) GetPost(channelId string, postId string, etag string) (*Result,
 	}
 }
 
+func (c *Client) GetPermalink(channelId string, postId string, etag string) (*Result, *AppError) {
```

Comment:
Would you be able to add a comment here, describing this function?

Something like:
```
// GetPermalink returns a permanent link to a post, based on the provided channel and post ID.
```

Also could you change the return type to be `(string, *ResponseData)` just like [this one](https://github.com/mattermost/platform/blob/master/model/client.go#L1357). We're trying to move the driver functions to use this new pattern and it would be much appreciated if you could change this one to match